sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -2270,7 +2270,8 @@ object GpuOverrides {
       }),
     exec[DataWritingCommandExec](
       "Writing data",
-      ExecChecks(TypeSig.commonCudfTypes, TypeSig.all),
+      ExecChecks((TypeSig.commonCudfTypes + TypeSig.NULL + TypeSig.STRUCT + TypeSig.MAP +
+        TypeSig.ARRAY + TypeSig.DECIMAL).nested(), TypeSig.all),
```

Comment:
what is the difference in this check and the file format specific ones?  is the schema at that point different than the plan output here?

Reply:
This check is useless in a way for `DataWritingCommandExec` as we are passing it for all datatypes and we will handle each type separately in the specific formats. 

Schema will be the same but we don't need to make the checks based on schema in most of the Execs therefore we are doing it inside the formats

Reply:
It's not useless, correct?  We need to pass this check before we pass the check for the scans, and this generates the documentation that we support decimal for this exec (with the caveat that it only works for some formats).  Speaking of caveats, this should use the `withPsNote` functionality in TypeChecks so we add the caveat that decimal type may not be supported by the specific output format.

Reply:
Yes, sorry, not useless at all. 

Great point for using `withPsNote`. I will make the change